lib/mongo/cursor.ex

```diff
@@ -8,6 +8,13 @@ import Record, only: [defrecordp: 2]
 defmodule Mongo.Cursor do
   @moduledoc false
 
+  @type t :: %__MODULE__{
+    conn: Mongo.conn,
+    coll: Mongo.collection,
+    query: BSON.document,
+    select: BSON.document | nil,
+    opts: Keyword.t
+  }
```

Comment:
The cursor modules are moduledoc private so I am thinking we should create a type `@opaque cursor :: Mongo.Cursor.t | Mongo.AggregationCursor.t | Mongo.SinglyCursor.t` on `Mongo` when referencing these types in public API.